app/models/notifications/mentioned_in_comment.rb

```diff
@@ -10,8 +10,15 @@ def deleted_translation_key
       "notifications.mentioned_in_comment_deleted"
     end
 
-    def self.filter_mentions(mentions, _mentionable, _recipient_user_ids)
-      mentions
+    def self.filter_mentions(mentions, mentionable, _recipient_user_ids)
+      ppl = mentionable.people_allowed_to_be_mentioned
+      if ppl == "all"
+        mentions
+      else
+        mentions.to_ary.delete_if {|mention|
```

Comment:
Well, the default scope is "All aspects" which means all contacts, but if it is fine with 10000, then I think it can hardly cause any issues.

I'd vote for readability and `where` clause then. If it cause any performance problems in future, it can be solved as a separate issue. I guess it comforms the Ruby/Rails way "features first, performance second".